lenskit/algorithms/als.py

```diff
@@ -465,8 +493,26 @@ def _train_iters(self, current, uctx, ictx):
             yield current
 
     def predict_for_user(self, user, items, ratings=None):
-        # look up user index
-        return self.score_by_ids(user, items)
+        if ratings is not None:
+            u_offset = None
+            if self.bias:
+                ratings, u_offset = self.bias.transform_user(ratings)
+
+            ri_idxes = self.item_index_.get_indexer_for(ratings.index)
+            ri_good = ri_idxes >= 0
+            ri_it = ri_idxes[ri_good]
+            ri_val = ratings.values[ri_good]
+
+            u_feat = _train_bias_row_lu(ri_it, ri_val, self.item_features_, self.regularization)
+            scores = self.score_by_ids(user, items, u_feat)
+
+            if self.bias:
+                return self.bias.inverse_transform_user(user, scores, u_offset)
+            else:
+                return scores
+        else:
+            # look up user index
+            return self.score_by_ids(user, items)
```

Comment:
Ok, so we aren't being consistent in de-biasing here. If `score_by_ids` de-biases, then when we have new ratings we debias twice; if it does not, then when we don't have new ratings we don't debias.